src/datamigration/azext_datamigration/manual/custom.py

```diff
@@ -0,0 +1,242 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=too-many-lines
+# pylint: disable=unused-argument
+# pylint: disable=line-too-long
+
+import ctypes
+import json
+import os
+import platform
+import subprocess
+import time
+import urllib.request
+from zipfile import ZipFile
+from knack.util import CLIError
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Common helper function to validate if the commands are running on Windows.
+# -----------------------------------------------------------------------------------------------------------------
+def validate_os_env():
+
+    if not platform.system().__contains__('Windows'):
+        raise CLIError("This command cannot be run in non-windows environment. Please run this command in Windows environment")
```

Comment:
CLIError will be deprecated. Please follow this [guideline](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#azure-cli-error-handling-guidelines)